x-pack/plugins/apm/public/components/fleet_integration/apm_policy_form/settings_definition/rum_settings.ts

```diff
@@ -4,10 +4,38 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
+import * as t from 'io-ts';
+import { either } from 'fp-ts/lib/Either';
 import { i18n } from '@kbn/i18n';
 import { PackagePolicyVars, SettingsRow } from '../typings';
 import { isSettingsFormValid, OPTIONAL_LABEL } from '../settings_form/utils';
 
+const arrayRegex = new RegExp(/[\[\]]/);
+function getAllowedOriginsRt() {
+  return new t.Type<string, string, unknown>(
+    'allowedOriginsRt',
+    t.string.is,
+    (input, context) => {
+      return either.chain(
+        t.string.validate(input, context),
+        (inputAsString) => {
+          return arrayRegex.test(inputAsString)
+            ? t.failure(
+                input,
+                context,
+                i18n.translate(
+                  'xpack.apm.fleet_integration.settings.rum.allowedHeadersValidation',
+                  { defaultMessage: 'Invalid value. Array is not allowed' }
```

Comment:
perhaps we should say something like "square brackets not allowed", since we're not specifically parsing for array syntax.

Reply:
sounds good to me.